### DISCUSSION TITLE
Example of VirtualizeOptions

### DISCUSSION THREAD

**Discussion by robalexclark** [Megabit/Blazorise#2694] [Q&A]
Please could we have an example of VirtualizeOptions, specifically setting the datagrid height?

Also, does Virtualize=true turn off Paging=true?

**Comment by stsrki** [comment]
To set the height use `FixedHeaderDataGridHeight="500px"`.

Yes, when Virtualize is enabled paging buttons are hidden. They would have little meaning when you scroll.

@David-Moreira can explain it better maybe

**Reply by robalexclark** [reply]
hmm Virtualize="true" FixedHeaderDataGridHeight="1200px" does not set the datagrid height to 1200px;

I can see that there are options in VirtualizeOptions to set the height intead but I can't quite figure out the syntax here.

**Comment by robalexclark** [comment]
VirtualizeOptions="@(()=> new VirtualizeOptions { DataGridHeight = "1200px" })" does not work because Cannot convert lambda expression to type 'VirtualizeOptions' because it is not a delegate type.

I can set VirtualizeOptions as a field and use that - is that how you are supposed to set it?

**Reply by David-Moreira** [reply]
Hello @robalexclark 

------------
>Also, does Virtualize=true turn off Paging=true?

Virtualize does turn off the paging as it has no utility for `Virtualize` feature, therefore makes no sense to handle or show. 
However, the `ButtonRow` will still display.

To add to this. If you want something similar but with paging, you can just set `FixedHeader` to true, and your desired height with `FixedHeaderDataGridHeight` and you'll be able to scroll while keeping the header fixed, and still mantain the Datagrid's inbuilt paging system.

The choice of keeping two separate heights for each feature, is that we might actually rework `Virtualize` in the future, and the height might end up working in a different way. Also, it could not be clear to a regular user that `FixedHeaderDataGridHeight` would have any effect in `Virtualize`.

----------

>VirtualizeOptions="@(()=> new VirtualizeOptions { DataGridHeight = "1200px" })" does not work because Cannot convert lambda expression to type 'VirtualizeOptions' because it is not a delegate type.
I can set VirtualizeOptions as a field and use that - is that how you are supposed to set it?

It's just a regular c# object you need to provide. So the syntax is like this:
`VirtualizeOptions='new(){ DataGridHeight = "1200px" }'`
You could also just initialize and bind a regular field or property to it, yes.

----------
I hope we've answered your questions @robalexclark and we'll definitely improve the examples, so you can just copy paste code in the future.... just let us know if you have any other questions. :)

But looking at you attempt to provide the options, makes me think if we should consider making it an `Action` or `Func`, on a future version, potentially making it more user friendly... What are your toughts about this, @stsrki ?

**Reply by stsrki** [reply]
Usually, I like `Action` or `Func` but in this case, I think it makes more sense to have an object. That is how we did so far for the options.
